config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        <module name="AvoidStarImport" />
+        <!-- We suppress this warning because the Java creators themselves disagree on this one, for example in the newest JUnit 5 -->
+        <module name="AvoidStaticImport">
+            <property name="severity" value="ignore" />
+        </module>
+        <module name="IllegalImport" />
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportControl"/> -->
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportOrder"/> -->
+        <module name="RedundantImport" />
+        <module name="UnusedImports" />
 
         <!-- Checks for Javadoc comments.                     -->
         <!-- See http://checkstyle.sourceforge.net/config_javadoc.html -->
         <module name="JavadocMethod">
             <property name="scope" value="package" />
-            <property name="allowMissingParamTags" value="true" />
+            <property name="allowMissingPropertyJavadoc" value="true" />
             <property name="allowMissingThrowsTags" value="true" />
-            <property name="allowMissingReturnTag" value="true" />
             <property name="allowThrowsTagsForSubclasses" value="true" />
             <property name="allowUndeclaredRTE" value="true" />
-            <property name="allowMissingPropertyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
         </module>
         <module name="JavadocType">
             <property name="scope" value="package" />
-            <property name="severity" value="ignore" />
         </module>
+        <!-- We ignore this warning, because variable should be self-explanatory, thus eliminating the need for Javadoc -->
         <module name="JavadocVariable">
-            <property name="scope" value="package" />
             <property name="severity" value="ignore" />
         </module>
         <module name="JavadocStyle">
-            <property name="checkEmptyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
+            <property name="checkFirstSentence" value="false" />
         </module>
 
+        <!-- Checks for Metrics. -->
+        <!-- See http://checkstyle.sourceforge.net/config_metrics.html -->
+        <module name="BooleanExpressionComplexity" />
+        <module name="ClassDataAbstractionCoupling" />
+        <module name="ClassFanOutComplexity" />
+        <module name="CyclomaticComplexity" />
+        <module name="JavaNCSS" />
+        <module name="NPathComplexity" />
+
+
+        <!-- Miscellaneous other checks.                   -->
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html -->
+        <module name="ArrayTypeStyle" />
+        <module name="AvoidEscapedUnicodeCharacters" />
+        <module name="CommentsIndentation" />
+        <module name="DescendantToken" />
```

Comment:
> WARNING: This is a very powerful and flexible check, but, at the same time, it is low-level and very implementation-dependent because its results depend on the grammar we use to build abstract syntax trees. Thus we recommend using other checks when they provide the desired functionality. Essentially, this check just works on the level of an abstract syntax tree and knows nothing about language structures.

Idk about this